department-of-veterans-affairs/next-build
# VACMS-10014 news story refactor

## Description

Closes #10014. Splits news story full and news story teaser into separate components. Fixes casing on these directories/files. Fixes pagination demo pages referencing teaser component. Fixes some typing related to person profile and news story. Updates storybook organization slightly to match proposed new component directory structure

## Testing done
- yarn dev, yarn storybook, yarn test

```sh
yarn test src/components/newsStory
yarn run v1.22.17
$ jest src/components/newsStory
info  - Loaded env from /Users/heffner/dev/next-build/.env.test
warn  - You have enabled experimental feature(s).
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.

warn  - SWC minify release candidate enabled. https://nextjs.org/docs/messages/swc-minify-enabled
warn  - SWC minify release candidate enabled. https://nextjs.org/docs/messages/swc-minify-enabled
 PASS  src/components/newsStoryTeaser/index.test.tsx
 PASS  src/components/newsStoryFull/index.test.tsx

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        2.213 s
Ran all test suites matching /src\/components\/newsStory/i.
✨  Done in 3.53s.
```

## Screenshots

![Screen Shot 2022-08-05 at 12 41 36 PM](https://user-images.githubusercontent.com/11279744/183152813-b9644706-9c49-4f19-91a3-7019eb331430.png)
![Screen Shot 2022-08-05 at 12 42 25 PM](https://user-images.githubusercontent.com/11279744/183152820-ebb6d82a-59d6-4082-9a20-8cc0fa722cbf.png)


## QA steps
- run `yarn dev`
- the news story full page, news story listing, and both pagination examples should all render correctly

- run `yarn storybook`, see that stories for both news story full and news story teaser exist and render correctly

### Definition of Done


### Is this PR blocked by another PR?
